python/cudf/cudf/io/csv.py

```diff
@@ -158,33 +162,48 @@ def read_csv(
             header = 0
 
     hex_cols: list[abc.Hashable] = []
-    new_dtypes: list[plc.DataType] | dict[abc.Hashable, plc.DataType] = []
+    cudf_dtypes: list[DtypeObj] | dict[abc.Hashable, DtypeObj] | DtypeObj = []
+    plc_dtypes: list[plc.DataType] | dict[abc.Hashable, plc.DataType] = []
```

Comment:
The dict key types are really this flexible? I guess these are anything that could be a pandas column "name"...

Reply:
Yeah since we also store column names in the `ColumnAccessor` dict keys, a column name just needs to be hashable, which also matches pandas.

(We have typing inconsistency in cudf classic too where sometimes column "names" are typed `str` (maybe mirroring libcudf?) when it should be `abc.Hashable`)

Reply:
Yeah we should resolve more of these inconsistencies eventually. My expectation is that once we add proper type stubs for pylibcudf and rework our internals around that we'll be able to work out better typing in our internals and slowly propagate that upwards. Right now so many deep parts of our code play fast and loose with types.